### DISCUSSION TITLE
hiding meteor showers is not saved in settings

### DISCUSSION THREAD

**Discussion by snoitaleR** [Stellarium/stellarium#906] [Q&A] [answered]
1. Hiding meteor showers [Ctrl]+[Shift]+[M].
2. Open the settings and save the view and settings [F2].
3. Restart Stellarium.
4. Meteor showers are displayed.

**Comment by gzotti** [comment, marked as answer]
This is a plugin-specific setting which must be configured in the plugin's settings dialog. (First option there.)

**Comment by snoitaleR** [comment]
thanks for the help.
an unobvious algorithm, but the goal is achieved,

**Comment by alex-w** [comment]
I guess it’s time to change a bit the behavior for saving the data.

**Comment by gzotti** [comment]
Just make sure to find a solution that does not assume properties for possibly inactive plugins are available. 
In this case, switching the GUI button could directly save the on/off flag.

**Comment by snoitaleR** [comment]
while working with the program, I can change a large number of parameters, but the next time I run it would be expensive to manually return all the parameters changed in the previous session.

**Comment by gzotti** [comment]
This is unclear: Do you prefer explicitly having to save preferences, or should the planetarium remember which settings you were running at time of the last quitting?

**Comment by snoitaleR** [comment]
the program has a default view and default settings. I wanted to make sure that by default meteor showers are not displayed when the program starts.

**Comment by gzotti** [comment]
Sure. This can be achieved by either disabling Meteor Showers and having to explicitly save settings (either like now in the plugin's config dialog, or by adding some functionality in the main config dialog's "Save Settings" that forwards this request to all plugins), or by just remembering the last state of the displayed buttons in the bottom bar. The question is what is the most user-friendly way. My personal preference would be "save last state automatically", but this is not what is currently happening.

**Comment by snoitaleR** [comment]
You can create a whole list of different configurations (not one as now), among which will be the configuration saved at the end of the last session of working with the program.

**Comment by gzotti** [comment]
You don't understand me.

**Comment by snoitaleR** [comment]
I just tried to combine both the configuration system (as it is now) and the recovery of the last session into a single system.
